config/routes.rb

```diff
@@ -1,8 +1,10 @@
 Rails.application.routes.draw do
-  devise_for :users
-  resources :organisations
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
   # Defines the root path route ("/")
   root "organisations#index"
+
+  devise_for :users
+
+  resources :organisations, except: [:show] do
+    resources :software_instances, shallow: true, path: :software
```

Comment:
See the [Rails Routing Guide](https://guides.rubyonrails.org/routing.html#shallow-nesting) for how shallow routing works.

Basically it allows one object to be nested inside another, but still be accessed directly when necessary. So for example, listing software instances is done within the context of the parent organisation so the url is `/organisations/:organisation_id/software_instances`, but if you have a software instance it can be view using its id alone: `/software_instances/:id`